features/dd-sdk-android-rum/src/test/kotlin/com/datadog/android/rum/internal/domain/scope/RumViewScopeTest.kt

```diff
@@ -7863,6 +7863,248 @@ internal class RumViewScopeTest {
+            val refreshRate = 1.0 / frameTime
+            refreshRates.add(refreshRate)
+            sum += refreshRate
+            min = kotlin.math.min(min, refreshRate)
+            max = kotlin.math.max(max, refreshRate)
+
+            testedScope.handleEvent(
+                RumRawEvent.UpdateExternalRefreshRate(frameTime),
+                mockWriter
+            )
+        }
+
+        val result = testedScope.handleEvent(
+            RumRawEvent.KeepAlive(),
+            mockWriter
+        )
+
+        // THEN
+        val expectedAverage = sum / refreshRates.size
+        argumentCaptor<ViewEvent> {
+            verify(mockWriter).write(eq(mockEventBatchWriter), capture(), eq(EventType.DEFAULT))
+            assertThat(lastValue)
+                .hasRefreshRateMetric(expectedAverage, min)
+        }
+        verifyNoMoreInteractions(mockWriter)
+        assertThat(result).isSameAs(testedScope)
+    }
+
+    @Test
+    fun `M ignore invalid frame time W handleEvent(UpdateExternalRefreshRate+KeepAlive) { zero frame time }`() {
+        // WHEN
+        testedScope.handleEvent(
+            RumRawEvent.UpdateExternalRefreshRate(0.0),
+            mockWriter
+        )
+        val result = testedScope.handleEvent(
+            RumRawEvent.KeepAlive(),
+            mockWriter
+        )
+
+        // THEN
+        argumentCaptor<ViewEvent> {
+            verify(mockWriter).write(eq(mockEventBatchWriter), capture(), eq(EventType.DEFAULT))
+            assertThat(lastValue)
+                .hasRefreshRateMetric(null, null)
+        }
+        verifyNoMoreInteractions(mockWriter)
+        assertThat(result).isSameAs(testedScope)
+    }
+
+    @Test
+    fun `M ignore invalid frame time W handleEvent(UpdateExternalRefreshRate+KeepAlive) { negative frame time }`(
+        forge: Forge
+    ) {
+        // GIVEN
+        val negativeFrameTime = -forge.aDouble(min = 0.001, max = 1.0)
+
+        // WHEN
+        testedScope.handleEvent(
+            RumRawEvent.UpdateExternalRefreshRate(negativeFrameTime),
+            mockWriter
+        )
+        val result = testedScope.handleEvent(
+            RumRawEvent.KeepAlive(),
+            mockWriter
+        )
+
+        // THEN
+        argumentCaptor<ViewEvent> {
+            verify(mockWriter).write(eq(mockEventBatchWriter), capture(), eq(EventType.DEFAULT))
+            assertThat(lastValue)
+                .hasRefreshRateMetric(null, null)
+        }
+        verifyNoMoreInteractions(mockWriter)
+        assertThat(result).isSameAs(testedScope)
+    }
+
+    @Test
+    fun `M prioritize external data W handleEvent(UpdateExternalRefreshRate+VitalUpdate+KeepAlive)`(
+        forge: Forge
+    ) {
+        // GIVEN
+        val externalFrameTime = forge.aDouble(min = 0.016, max = 0.017) // ~60 FPS
+        val expectedExternalRefreshRate = 1.0 / externalFrameTime
+
+        val internalRefreshRate = forge.aDouble(min = 30.0, max = 45.0) // Different range
```

Comment:
probably we can put some very wide range here since it is about priority? so range doesn't probably matter. Otherwise `min = 0.016, max = 0.017` looks a bit questionable.

Same can be applied for the tests below.